pkg/registries/registries.go

```diff
@@ -139,7 +166,12 @@ func EditRegistriesConfig(config *sysregistriesv2.V2RegistriesConf, insecureRegi
 		reg.Insecure = true
 		for i := range config.Registries {
 			reg := &config.Registries[i]
-			if scopeMatchesRegistry(reg.Location, insecureReg) {
+			scope := reg.Location
+			// If Prefix is a valid wildcard entry, we set scope to Prefix
+			if isValidWildcard(reg.Prefix) {
```

Comment:
Like above, always use `Prefix` if non-empty, wildcard or not. Also for blocked.

Reply:
Fixed